Refactor clustering (#1728)
emqtt/emq-docs-cn: en_US/deploy/cluster/lb.md

# Configure Load Balancer

> Load Balancer (LB) will balance the load among multiple network components, optimize resource usage, and avoid system malfunctions caused by overload. LB is not a mandatory component in EMQX, but could bring some obvious system benefits, for example:

Reviewer: ```suggestion
Load Balancer (LB) balances the load among multiple network components and optimizes resource usage to avoid system malfunctions caused by overload. LB is not a mandatory component in EMQX, but it can bring some obvious system benefits, for example:
```

- Balance the load of EMQX to avoid single node overload;
- Simplify client configuration, the client only needs to connect to the LB and need not worry about the scaling within the cluster;
- TLS/SSL termination, to reduce the load of EMQX clusters;
- Improve cluster security, with LB configured at the front end of the cluster, unwanted traffic can be blocked to protect the EMQX cluster from malicious attacks. 

This chapter introduces how to configure LB in EMQX. 

## Architecture

For an EMQX cluster configured with LB, the LB will handle the incoming TCP traffic and then distribute the received MQTT connection requests and messages to different EMQX nodes. The typical deployment architecture is as follows:

<img src="./assets/lb_1.png" alt="image" style="zoom:40%;" />

If SSL/TLS is enabled, it is recommended to terminate the SSL/TLS connection at LB, that is, to use SSL/TLS to secure the connection between clients and LB and then use TCP connection between LB and EMQX nodes, maximizing the performance of the EMQX cluster. The architecture is as follows:

> <img src="./assets/lb_2.png" alt="TLS termination" style="zoom:45%;" />

Reviewer: 文字里说clients和LB之间是用SSL/TLS连接，但是图上是TCP。我记得之前有一张图是显示SSL/TLS的。

| [HAProxy](https://www.haproxy.org) | Yes             | <https://www.haproxy.com/solutions/load-balancing.html> |
| [NGINX](https://www.nginx.com)     | Yes             | <https://www.nginx.com/solutions/load-balancing/>       |

> The following section will take the HAProxy or NGINX as an example to illustrate how to configure an LB in EMQX cluster. 

Author: ```suggestion
The following section takes the HAProxy or NGINX as an example to illustrate how to configure an LB in EMQX cluster. 
```
